lib/utils/cache.js

```diff
@@ -1,96 +1,57 @@
-const fs = require("fs");
-const tmpdir = require("os").tmpdir;
+const createDebugLogger = require("debug");
+const { flow, get } = require("lodash/fp");
+const Database = require("better-sqlite3");
 
-const pify = require("pify");
-const mkdirp = pify(require("mkdirp"));
-const toPairs = require("lodash/fp/toPairs");
+const SafeJSON = require("./safe-json");
 
-const DEFAULT_FILENAME = "default-cache.json";
+const DEFAULT_DATABASE = ".inspectpack-cache.db";
+const DEFAULT_NAME = "inspectpack-cache";
 
-const readFile = pify(fs.readFile);
-const writeFile = pify(fs.writeFile);
-const access = pify(fs.access);
-
-// Function adapted from:
-// http://2ality.com/2015/08/es6-map-json.html#converting-a-string-map-to-and-from-an-object
-function serializeMap(stringKeyedMap) {
-  const plainObject = Object.create(null);
-  for (const pair of stringKeyedMap.entries()) {
-    const key = pair[0];
-    const value = pair[1];
-    plainObject[key] = value;
-  }
-  return JSON.stringify(plainObject);
-}
-
-// Adapted from interlock with permission:
-// https://github.com/interlockjs/interlock/blob/master/src/optimizations/file-cache/index.js
-function getCacheFilePath(opts) {
-  const canonicalPath = opts.cacheDir || tmpdir();
-
-  const initPath = opts.cacheDir
-    ? mkdirp(canonicalPath)
-    : Promise.resolve();
-
-  return initPath
-    // eslint-disable-next-line no-bitwise
-    .then(() => access(canonicalPath, fs.R_OK | fs.W_OK))
-    .then(() =>
-      path.join(
-        canonicalPath,
-        opts.scope || DEFAULT_FILENAME
-      ))
-    .catch(() => {
-      throw new Error(
-        `Unable to access cache directory.
-        Please check the directory's user permissions:
-        ${canonicalPath}`
-      );
-    });
-}
+const CREATE_TABLE_QUERY =
+  "CREATE TABLE IF NOT EXISTS cache (key TEXT PRIMARY KEY, value TEXT);";
+const GET_QUERY = "SELECT value FROM cache WHERE key IS $key;";
+const SET_QUERY = "INSERT OR REPLACE INTO cache (key, value) VALUES ($key, $value);";
 
 module.exports = class Cache {
   /**
    * Asynchronously create a new cache from disk.
    *
-   * @param   {Object} opts                  Object options
-   * @param   {Object} opts.scope            The prefix for this cache's file
-   * @param   {Object} opts.cacheDir         The directory to store cache files
-   * @returns {Promise<Cache>}               The hydrated cache
+   * @param   {Object}   opts           Object options
+   * @param   {String}   opts.filename  The filename of the SQLite database
+   * @param   {String}   opts.name      A name for this instance used in debugging
+   * @returns {Cache}                   The hydrated cache
    */
-  static init(opts) {
-    opts = opts || {};
-
-    return getCacheFilePath(opts)
-      .then(filePath => readFile(filePath, "utf8"))
-      .then(file => new Cache(opts, new Map(toPairs(JSON.parse(file)))))
-      .catch(() => new Cache(opts));
+  static create(opts = {}) {
+    const log = createDebugLogger(
```

Comment:
Do we need to introduce a new dependency and full logger just for the cache?

Reply:
I had a train of thought that we could log periodic cache stats along with errors under a debug flag, but that train long left the station and derailed. I'd be happy to axe it–do we want to not log at all?